packages/html-reporter/src/links.tsx

```diff
@@ -113,3 +113,36 @@ export function generateTraceUrl(traces: TestAttachment[]) {
 }
 
 const kMissingContentType = 'x-playwright/missing';
+
+type AnchorID = string | ((id: string | null) => boolean) | undefined;
+
+export function useAnchor(id: AnchorID, onChange: (isRevealed: boolean) => void) {
+  React.useEffect(() => {
+    const listener = () => {
+      const params = new URLSearchParams(window.location.hash.slice(1));
+      const anchor = params.get('anchor');
+      const isRevealed = typeof id === 'function' ? id(anchor) : anchor === id;
+      onChange(isRevealed);
+    };
+    window.addEventListener('popstate', listener);
+    return () => window.removeEventListener('popstate', listener);
+  }, [id, onChange]);
+}
+
+export function Anchor({ id, onChange, children }: React.PropsWithChildren<{ id: AnchorID, onChange?(isRevealed: boolean, ref: HTMLDivElement): void }>) {
+  const ref = React.useRef<HTMLDivElement>(null);
+  const onAnchorChange = React.useCallback((isRevealed: boolean) => {
+    if (!ref.current)
+      return;
+
+    const preventDefault = onChange?.(isRevealed, ref.current);
+    if (preventDefault)
```

Comment:
- This `preventDefault` logic is unused. Let's drop it for now?
- Same goes for the `ref` parameter.
- Why would anyone care about `!isRevealed` situation? If anything, I'd call this callback `onReveal()` and call only on reveal.

Reply:
I tried to be cheeky and support every thinkable usecase. But I guess we can remove the unused stuff and add it once needed :)